tests/detectors/test_detectors_fileformats.py

```diff
@@ -0,0 +1,131 @@
+
+
+def test_fileispickled_valid_format():
+    d = garak.detectors.fileformats.FileIsPickled()
+    file_attempt = Attempt(prompt="")
+    file_attempt.notes["format"] = "local filename"
+    file_attempt.outputs = [""]
+    l = list(d.detect(file_attempt))
+    assert l == []
+
+
+def test_fileispickled_nonpickle():
+    d = garak.detectors.fileformats.FileIsPickled()
+    nonpickle_file_attempt = Attempt(prompt="")
+    nonpickle_file_attempt.notes["format"] = "local filename"
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as nonpickle_file:
+        nonpickle_file.write(
+            "Do you know what the sad part is, Odo? I'm a very good tailor."
+        )
+        nonpickle_file.close()
+        nonpickle_file_attempt.outputs = [nonpickle_file.name]
+        l = list(d.detect(nonpickle_file_attempt))
+        assert l == [0.0]
+
+
+def test_fileispickled_pickle_default():
+    d = garak.detectors.fileformats.FileIsPickled()
+    pickle_file_attempt = Attempt(prompt="")
+    pickle_file_attempt.notes["format"] = "local filename"
+    with tempfile.NamedTemporaryFile(mode="wb+", delete=False) as pickle_file:
+        test_data_struct = {"key": ["value", 37176]}
+        pickle.dump(test_data_struct, pickle_file)
+        pickle_file.close()
+        pickle_file_attempt.outputs = [pickle_file.name]
+        l = list(d.detect(pickle_file_attempt))
+        assert l == [1.0]
+
+
+pickle_versions = [0, 1, 2, 3, 4, 5]
+
+
+@pytest.mark.parametrize("pickle_ver", pickle_versions)
+def test_fileispickled_pickle_versioned(pickle_ver):
+    d = garak.detectors.fileformats.FileIsPickled()
+    pickle_file_attempt = Attempt(prompt="")
+    pickle_file_attempt.notes["format"] = "local filename"
+    with tempfile.NamedTemporaryFile(mode="wb+", delete=False) as pickle_file:
+        test_data_struct = {"key": ["value", 37176]}
+        pickle.Pickler(pickle_file, pickle_ver).dump(test_data_struct)
+        pickle_file.close()
+        pickle_file_attempt.outputs = [pickle_file.name]
+        l = list(d.detect(pickle_file_attempt))
+        assert l == [1.0]
+
+
+def test_fileisexectuable_nonexec():
+    exec_attempt = garak.attempt.Attempt(prompt="")
+    exec_attempt.notes["format"] = "local filename"
+    d = garak.detectors.fileformats.FileIsExecutable()
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as test_file:
+        test_file.write("I'm a very good tailor.")
+        test_file.close()
+        exec_attempt.outputs = [test_file.name]
+        results = d.detect(exec_attempt)
+        assert list(results) == [0.0]
+
+
+def test_fileisexectuable_exec():
+    exec_attempt = garak.attempt.Attempt(prompt="")
+    exec_attempt.notes["format"] = "local filename"
+    path_prefix = Path("tests/resources/fileformats/exec_files/")
+
+    encoded_exec_filenames = glob.glob(str(path_prefix / "*.base64"))
+    exec_filenames = []
+    for encoded_exec_filename in encoded_exec_filenames:
```

Comment:
The decoding process here may be better implemented as a `fixture` that registers the file to be deleted post test, as is a failure of the `assert` would leave all decoded files on disk in the temporary location.

Untested possible refactor:

``` Python
path_prefix = Path("tests/resources/fileformats/exec_files/")

@pytest.fixture
def decoded_file(request, encoded_exec_filename):
    with open(encoded_exec_filename, "r", encoding="utf-8") as encodedfile:
        with tempfile.NamedTemporaryFile(mode="wb", delete=False) as binfile:
            binfile.write(binascii.a2b_base64(encodedfile.read()))
            binfile.close()
            exec_filenames.append(binfile.name)

    def remove_decoded():
        with contextlib.suppress(FileNotFoundError):
            os.remove(binfile.name)

    return binfile.name
```
``` Python
@pytest.mark.parametrize("encoded_exec_filename", glob.glob(str(path_prefix / "*.base64")))
def test_fileisexectuable_exec(decoded_file):
    d = garak.detectors.fileformats.FileIsExecutable()
    exec_attempt.outputs = list(decoded_file)
    results = d.detect(exec_attempt)
    assert list(results) == [1.0]
```

Reply:
yup, implemented